jax/_src/core.py

```diff
@@ -594,24 +594,29 @@ class Tracer(typing.Array):
   __array_priority__ = 1000
   __slots__ = ['_trace', '_line_info']
 
+  def _error_repr(self):
+    if hasattr(self, "shape") and hasattr(self, "dtype"):
+      return f"traced array with shape {self.dtype}{list(self.shape)}"
+    return f"traced array with aval {self.aval}"
```

Comment:
What do you think of calling `aval.str_short()` in place of this? I believe it recovers what this function would do in the case of a `ShapedArray`.

Reply:
Tried this – the problem is that in the case of concrete tracers, `str_short` prints the full repr of the concrete array it traces: https://github.com/google/jax/blob/bfe8acb31e04a540daad3f568239ec0e5c3f0d0f/jax/_src/core.py#L1605

Reply:
Maybe we should change this behavior, since we want `str_short` to produce a short string? @mattjj

Reply:
How about `core.raise_to_shaped(t.aval).str_short()` then?

Reply:
Sure, sounds good